metakb/transform/civic.py

```diff
@@ -0,0 +1,765 @@
+
+        :return: A list of dictionaries containing transformations to CDM.
+        """
+        data = self._extract()
+        responses = list()
+        evidence_items = data['evidence']
+        assertions = data['assertions']
+        variants = data['variants']
+        genes = data['genes']
+        cdm_evidence_items = dict()  # EIDs that have been transformed to CDM
+        propositions_documents_ix = {
+            'document_index': 1,  # Keep track of document index value
+            'documents': dict(),  # {document_id: document_index}
+            'proposition_index': 1,  # Keep track of proposition index value
+            'propositions': dict()  # {tuple: proposition_index}
+        }
+        self._transform_evidence(responses, evidence_items, variants, genes,
+                                 propositions_documents_ix,
+                                 cdm_evidence_items)
+        self._transform_assertions(responses, assertions, variants, genes,
+                                   cdm_evidence_items,
+                                   propositions_documents_ix)
+        return responses
+
+    def _transform_evidence(self, responses, evidence_items, variants, genes,
+                            propositions_documents_ix, cdm_evidence_items):
+        """Add transformed EIDs to the response list.
+
+        :param list responses: A list of dicts containing CDM data
+        :param list evidence_items: A list of CIViC evidence items
+        :param dict variants: A dict of CIViC variant records
+        :param dict genes: A dict of CIViC gene records
+        :param dict propositions_documents_ix: Keeps track of proposition and
+            document indexes
+        :param dict cdm_evidence_items: A dict containing evidence items that
+            have been transformed to the CDM
+        """
+        for evidence in evidence_items:
+            descriptors = self._get_descriptors(evidence, genes, variants)
+            if not descriptors:
+                continue
+            else:
+                therapy_descriptors, variation_descriptors, disease_descriptors = descriptors  # noqa: E501
+
+            propositions = \
+                self._get_tr_propositions(evidence, variation_descriptors,
+                                          disease_descriptors,
+                                          therapy_descriptors,
+                                          propositions_documents_ix)
+
+            # We only want therapeutic response for now
+            if not propositions:
+                continue
+
+            gene_descriptors = self._get_gene_descriptors(
+                self._get_record(evidence['gene_id'], genes))
+
+            documents = self._get_evidence_document(evidence['source'],
+                                                    propositions_documents_ix)
+
+            assertion_methods = [schemas.AssertionMethod(
+                id='assertion_method:001',
```

Comment:
I'm guessing we will a) need some kind of method to create unique ID numbers and/or b) namespace this somehow? So that eg we don't have conflicts between assertion methods tied to different sources, etc

Reply:
Good catch. I can add something to `schemas.py` to store these constants for now.

Reply:
Added in [this commit.](https://github.com/cancervariants/metakb/commit/1be7dcdf94dd5d95817a9dfbec22408945a7c544)

Reply:
Okay, I misremembered what the Assertion Method class is used for, but I agree that declaring some constants is the way to go forward (especially for people like me who commit typos at a rate of once per minute)